das/worker.go

```diff
@@ -0,0 +1,97 @@
+package das
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/header"
+
+	"go.uber.org/multierr"
+)
+
+type worker struct {
+	lock  sync.Mutex
+	state workerState
+}
+
+// workerState contains important information about the state of a
+// current sampling routine.
+type workerState struct {
+	job
+
+	Curr   uint64 `json:"current"`
+	Err    error  `json:"error,omitempty"`
+	failed []uint64
+}
+
+type job struct {
```

Comment:
A bit off that only some structs are documented while others aren't

Reply:
showed some love to other structs by adding comments